### DISCUSSION TITLE
Energiewerte Wallbox

### DISCUSSION THREAD

**Discussion by Tom88s** [torbennehmer/hacs-e3dc#178] [Help]
Wie werden die Energiewerte der Wallbox berechnet?
Ich glaub bei mir passt da etwas nicht…

Es geht um die Energiewerte der Wallbox
Energy all
Energy net
Energy sun

Vielen Dank vorab

Thomas

**Comment by bullitt186** [comment]
Hi Thomas, I just had a look into it.
The values "energy net" / "energy sun" come directly from the Hauskraftwerk. "energy all" is just the sum of the two sensors.
However in the source code of the underlying library [python-e3dc](https://github.com/fsantini/python-e3dc), it states that these values are monthly, so they are resetted periodically.

Further, i checked the behaviour on my setup, where i have the version with wallbox support already running for a while.
In fact the value for "sun" is being reset (and consequently "all" takes a drop as well), while "net" is not reset. See image. I don't know why the Hauskraftwerk behaviour is like this, i also don't know why the reset is not on the month-change. maybe my clock  in the Hauskraftwerk is off? 
<img width="985" alt="image" src="https://github.com/user-attachments/assets/dadf1322-77e2-469e-b3bc-a4192bd46b3e">

To overcome this limitation / behaviour from the Hauskraftwerk, I suggest to set up [utility meter helpers](https://www.home-
assistant.io/integrations/utility_meter/), with regular reset activated (see screenshot). These meters should only grow and not being reset. I just set up three meters and will observe.
<img width="576" alt="image" src="https://github.com/user-attachments/assets/fffad405-3828-4a1c-bdc1-5a1e665f3e26">

By the way you can set these utility meter values via action "[calibrate](https://www.home-assistant.io/integrations/utility_meter/#action-utility_metercalibrate)"

**Reply by ga63cez** [reply]
Hi! Any news on this one? Did the workaround work fine for you? I've got a similar problem but for me the resets seem to be completely random...

![image](https://github.com/user-attachments/assets/df0ee965-0ed5-455f-897e-f55f3f8ebf5b)

Cheers!
